awx/ui/client/features/projects/projectsList.controller.js

```diff
@@ -0,0 +1,376 @@
+                    // Use the Refresh button to monitor the status.', 'alert-info');
+                } else {
+                    ProjectUpdate({ scope: $scope, project_id: project.id });
+                }
+            }
+        });
+    };
+
+    function buildTooltips (project) {
+        project.statusIcon = GetProjectIcon(project.status);
+        project.statusTip = GetProjectToolTip(project.status);
+        project.scm_update_tooltip = vm.strings.get('update.GET_LATEST');
+        project.scm_update_disabled = false;
+
+        if (project.status === 'pending' || project.status === 'waiting') {
+            project.scm_update_disabled = true;
+        }
+
+        if (project.status === 'failed' && project.summary_fields.last_update && project.summary_fields.last_update.status === 'canceled') {
+            project.statusTip = vm.strings.get('status.CANCEL');
+            project.scm_update_disabled = true;
+        }
+
+        if (project.status === 'running' || project.status === 'updating') {
+            project.scm_update_tooltip = vm.strings.get('udpate.UPDATE_RUNNING');
+            project.scm_update_disabled = true;
+        }
+
+        if (project.scm_type === 'manual') {
+            project.statusIcon = 'none';
+            project.statusTip = vm.strings.get('status.NOT_CONFIG');
+            project.scm_update_tooltip = vm.strings.get('update.MANUAL_PROJECT_NO_UPDATE');
+            project.scm_update_disabled = true;
+        }
+    }
+
+    function cancelSCMUpdate (projectData) {
+        Rest.setUrl(projectData.related.current_update);
+        Rest.get()
+            .then(({ data }) => {
+                $scope.$emit('Check_Cancel', data);
+            })
+            .catch(createErrorHandler(projectData.related.current_update, 'GET'));
+    }
+
+    function reloadList () {
+        Wait('start');
+        const path = GetBasePath(vm.list.basePath) || GetBasePath(vm.list.name);
+        qs.search(path, $state.params.project_search)
+            .then((searchResponse) => {
+                vm.dataset = searchResponse.data;
+                vm.projects = vm.dataset.results;
+            })
+            .finally(() => Wait('stop'));
+    }
+
+    function createErrorHandler (path, action) {
+        return ({ data, status }) => {
+            const hdr = strings.get('error.HEADER');
+            const msg = strings.get('error.CALL', { path, action, status });
+            ProcessErrors($scope, data, status, null, { hdr, msg });
+        };
+    }
+}
+
+projectsListController.$inject = [
+    '$filter',
+    '$scope',
+    '$rootScope',
+    '$state',
+    '$log',
+    'Dataset',
+    'Find',
+    'Empty',
+    'Alert',
+    'Rest',
+    'ProcessErrors',
+    'GetProjectIcon',
+    'GetProjectToolTip',
+    'resolvedModels',
+    'ProjectsStrings',
+    'Wait',
+    'ngToast',
+    'Prompt',
+    'GetBasePath',
+    'QuerySet',
+    'ProjectUpdate',
+];
```

Comment:
I think it is worth the investment to try to remove as many of these dependencies as we can. Are any of this easily removable, replaceable with helper functions, etc.?

Reply:
`GetProjectIcon` and `GetProjectToolTip` might be easy targets here - https://github.com/ansible/awx/search?q=GetProjectIcon&unscoped_q=GetProjectIcon

Maybe this logic could live in some sort of component?

Reply:
Yes you're right... I found the original buildtoolTip logic to be very confusing...